### DISCUSSION TITLE
What if I **do** want 24/7 recordings?

### DISCUSSION THREAD

**Discussion by natevw** [blakeblackshear/frigate#8295] [Ask A Question] [answered]
I'm not a long-time Frigate user, in fact I just started and so the current documentation for recording is kind of confusing.

There's a question "What if I don't want 24/7 recordings?" and it gives an example saying "If you only used clips in previous versions with recordings disabled, you can use the following config to get the same behavior":

```
record:
  enabled: True
  events:
    retain:
      default: 10
```

but then it says "This is also the default behavior when recordings are enabled." Mmmkay…wait? So basically if I don't want 24/7 recordings then not only is there an example but I don't even need to use that because it's already the default.

But… what if I **do** want 24/7 recordings. Later on in the documentation it says:

> Frigate saves from the stream with the record role in 10 second segments.

that sounds great, but it seems the default event thing takes precedence. That is, for cameras with **either** the `- detect` role *or* if I add the `- record` role too, I just get the same thing: random-length clips because it tripped the object detection.

I expected that when I added the `- record` role I would get this behavior mentioned:

> Frigate saves from the stream with the record role in 10 second segments.

But I don't seem to really get that. Sometimes I'll get a long clip for some reason like it just reaaaaaaaaaaaaally got interested our parked car for ten minutes, and even then it might *also* records separate overlapping copy files for some *other* object it detects during that same time period? But never 10sec segments that can get retained/dropped on their own.

What I imagined (and wouldn't mind having):

```
|   10secs   |   10secs   |   10secs   |   10secs   |   10secs   |   10secs   |   10secs   |   10secs   |   10secs   |   10secs   | …
                      [some event]                                                 [ some other event later]
             |  retained  |  retained  |                                      |  retained  |  retained  |  retained  |
```

What I get instead:

```
nothing…        | 19m 42s recording because it 79% totally thought a piece of firewood was an "In Progress" person all afternoon |            …nothing…     | 18s some 73% car does drive by |     …nothing…
                          | 1m 52s for a 72% car zipping through mid-clip |
```

How *do* I get what sounds like might still be an option (and maybe was the only option on prior versions?), just a bunch of continuous back-to-back 10 seconds segments everything but then "retain recording segments that overlap with events"? Right now it seems like I just get segments directly and *only* if tied to events and however long Frigate thinks the events themselves last.

**Comment by NickM-27** [comment, marked as answer]
It definitely seems like some things could be explained better. 

This 19 m 42s recording you're referring to made up of the 10 second recording segments. Every recording clip is just generated using these recording segments. 

The examples below show it with 24/7 recording (in this case motion based)

```

record:
  enabled: True
  retain:
    days: 7
    mode: motion
  events:
    retain:
      default: 14
      mode: active_objects
```

**Reply by natevw** [reply]
Thanks! Seems like the main starting point is to have the base `record.retain.days` be non-zero? Just found this section in the main annotated config file page:

```
  retain:
    # Optional: Number of days to retain recordings regardless of events (default: shown below)
    # NOTE: This should be set to 0 and retention should be defined in events section below
    #       if you only want to retain recordings of events.
    days: 0
```

so I could leave mode as `all` (or as you have, `motion` might be a nice compromise) but main first thing is to up the `days: N` as you have it before anything will record beyond events it seems. I'd seen that the file lengths in the raw folder varied and just assumed they were 1:1 corresponding to what got shown/downloaded through the web UI, but didn't look carefully enough to notice that they were all still ± a certain small average amount probably just varying a bit due to compression.
